frontend/src/features/payment/stripe/components/StripePaymentBlock.tsx

```diff
@@ -77,18 +74,26 @@ const StripeCheckoutForm = ({
       //`Elements` instance that was used to create the Payment Element
       elements,
       confirmParams: {
-        return_url,
+        // 1. Responders should return back to the same payment page to view the receipt
+        // 2. Your customer will be redirected to your `return_url`. For some payment
+        //    methods like iDEAL, your customer will be redirected to an intermediate
+        //    site first to authorize the payment, then redirected to the `return_url`.
+        return_url: window.location.href,
       },
+
+      // The default is `redirect?: 'always'` and this will introduce
+      // a bad UX of triggering a page reload of the same page.
+      // By setting if_required we can control when it reloads,
+      // which is after the payment had completed successfully.
+      redirect: 'if_required',
     })
 
     if (result.error && result.error.message) {
-      // Show error to your customer (for example, payment details incomplete)
       setStripeMessage(result.error.message)
     } else {
-      setStripeMessage('')
-      // Your customer will be redirected to your `return_url`. For some payment
-      // methods like iDEAL, your customer will be redirected to an intermediate
-      // site first to authorize the payment, then redirected to the `return_url`.
+      // In the event that customer is not on a payment that has a redirected flow,
+      // we will trigger a payment status refetch
+      triggerPaymentStatusRefetch()
```

Comment:
should we still keep the `setStripeMessage('')`, so that we keep the state consistent?

Reply:
It's reset on [L65](https://github.com/opengovsg/FormSG/pull/6001/files/ff6bab87ece2f0d878b03c8d145fd8876e3dc14c#diff-abbd02937e4bd0ccb8beaacd056226b8e3b9bfa0faf9f18db916e04710dd3dccR65).

I was thinking that we should clear the error message on `Submit Payment` button press so that the previous error message, if any, would be cleared. If not, the responder will be still seeing the previous error message even though they've updated their payment information and clicked `Submit Payment`.

Reply:
good point! i missed that line, thanks for the clarification 👍